src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/_params.py

```diff
@@ -59,6 +70,26 @@ def completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-a
         return list(getattr(client, func)(container_name))
     return completer
 
+def dir_path_completer(prefix, action, parsed_args, **kwargs): # pylint: disable=unused-argument
+    client = _get_client(FileService, parsed_args)
+    share_name = parsed_args.share_name
+    directory_name = ''
+    if prefix:
+        directory_name = prefix
+        while directory_name:
+            if not client.exists(share_name, directory_name):
```

Comment:
No, but I can cut the looping short. Right now it will keep trimming sections on \, but realistically it only needs to do that once. The case is where you have dir/part where part is, say, a partial filename. It will first see if "dir/part" is a valid directory, because if it is, you want to list the files in that directory. In this case, where it is not, the first call would fail. It would then see if "dir/" is a directory, which it is, and list the files. Thus you'd get your filename completed. 

However if you tried to complete on a/b/c/foo, if a/b/c isn't a valid directory, there's no value in checking a/b or a/.

---
In reply to: [75172565](https://github.com/Azure/azure-cli/pull/702#discussion_r75172565) [](ancestors = 75172565)